scripts/globals/campaign.lua

```diff
@@ -10,6 +10,133 @@ PAST_SANDORIA = 5;
 PAST_BASTOK = 6;
 PAST_WINDURST = 7;
 
+-- -------------------------------------------------------------------
+-- getMedalRank()
+-- Returns the numerical Campaign Medal of the player.
+-- -------------------------------------------------------------------
+
+function getMedalRank(player)
+    local rank = 0;
+    local medals =
+    {
+         0x039C, 0x039D, 0x039E, 0x039F, 0x03A0, 0x03A1, 0x03A2,
+         0x03A3, 0x03A4, 0x03A5, 0x03A6, 0x03A7, 0x03A8, 0x03A9,
+         0x03AA, 0x03AB, 0x03AC, 0x03AD, 0x03AE, 0x03AF
+    }
+    while (player:hasKeyItem(medals[rank + 1]) == true) do
+        rank = rank + 1;
+    end;
+    return rank;
+end;
+
+-- -------------------------------------------------------------------
+-- get[nation]NotesItem()
+-- Returns the item ID and cost of the Allied Notes indexed item
+-- (the same value as that used by the vendor event.)
+-- -------------------------------------------------------------------
+
+-- Format:
+-- ListName_AN_item[optionID] = itemID; -- ItemName
+-- ListName_AN_price[optionID] = cost; -- ItemName
+
+local SandOria_AN_item = {}
+
+SandOria_AN_item[2] = 15754; -- Sprinter's Shoes
+SandOria_AN_item[258] = 5428; -- Scroll of Instant Retrace
+SandOria_AN_item[514] = 14584; -- Iron Ram jack coat
+SandOria_AN_item[770] = 14587; -- Pilgrim Tunica
+SandOria_AN_item[1026] = 16172; -- Iron Ram Shield
+SandOria_AN_item[1282] = 15841; -- Recall Ring: Jugner
+SandOria_AN_item[1538] = 15842; -- Recall Ring: Pashow
+SandOria_AN_item[1794] = 15843; -- Recall Ring: Meriphataud
+SandOria_AN_item[2050] = 10116; -- Cipher: Valaineral
+
+local SandOria_AN_price = {}
+
+SandOria_AN_price[2] = 980; -- Sprinter's Shoes
+SandOria_AN_price[258] = 10; -- Scroll of Instant Retrace
+SandOria_AN_price[514] = 1500; -- Iron Ram jack coat
+SandOria_AN_price[770] = 1500; -- Pilgrim Tunica
+SandOria_AN_price[1026] = 4500;-- Iron Ram Shield
+SandOria_AN_price[1282] = 5000;-- Recall Ring: Jugner
+SandOria_AN_price[1538] = 5000;-- Recall Ring: Pashow
+SandOria_AN_price[1794] = 5000;-- Recall Ring: Meriphataud
+SandOria_AN_price[2050] = 2000;-- Cipher: Valaineral
+
+function getSandOriaNotesItem(i)
+    return SandOria_AN_item[i],SandOria_AN_price [i];
+end;
+
+local Bastok_AN_item = {}
+
+Bastok_AN_item[2] = 15754; -- Sprinter's Shoes

```

Comment:
i think it's easier to understand if you just do 0,1,2.. etc (it's just bit.rshift(option,8)) instead of 2,258, etc

also, combining the price and id in the same table is also a good idea
Bastok_AN_item = { {id=xxx, price=xxx},{  ... etc